coding-the-legislation/35_periods.md

```diff
@@ -143,4 +191,29 @@ class salary(Variable):
         ...
 ```
 
-We can now provide an input for `2015` for `salary`: no error will be raised, and the value will be automatically split between the 12 months of `2015.
+We can now provide an input for `2015` for `salary`: no error will be raised, and the value will be automatically split between the 12 months of `2015`.
+
+## How to use Instants to represent cutoff dates.
+
+Sometimes, some results depend on a value calculted at a specific date.
+Instants can be used to calculate the value of a parameter or a variable at such date, for example, a scholarship that is awarded only if the student is yonger than 25 on july first of the same year:
+
+```py
+from openfisca_core.periods import Instant
+
+class eligible_for_scholarship(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        july_first = Instant(
+            (period.start.year, 7, 1)
+        ).period('month')
+
+        age_condition = person('age', july_first) <= 25
```

Comment:
This has been tested :)
I tested it on the country package and I got the inspiration from
`aide_logement_base_ressources_eval_forfaitaire` where I found: 
```py
 def eval_forfaitaire_tns():
            last_july_first = Instant(
                (period.start.year if period.start.month >= 7 else period.start.year - 1,
                7, 1))
            smic_horaire_brut = parameters(last_july_first).cotsoc.gen.smic_h_b
            travailleur_non_salarie_i = famille.members('travailleur_non_salarie', period)
            any_tns = famille.any(travailleur_non_salarie_i)
            return any_tns * 1500 * smic_horaire_brut
```

Reply:
Ok, just got it, I didn't see the `.period('month')` line 211. That why it doesn't crash.

But that makes this example quite misleading as `july_first` here is not a day, but the month of july. And we are not calculating `age` for an _instant_, but, like as usual, for a _period_.

Also, regarding the use and import of the `Instant` constructor here, I could see several simpler ways of reaching the same result  :

```py
july = "{}-07".format(period.start.year)
age_condition = person('age', july) <= 25
```

```py
july = period.this_year.first_month.offset(6, MONTH)
age_condition = person('age', july) <= 25
```

Reply:
So, to sum up, calculating variables for an instant instead of a period is probably [a relevant thing to think about](https://github.com/openfisca/openfisca-doc/pull/160#discussion_r214749951), but OpenFisca doesn't really allow it now. 

I'd therefore remove this section, to avoid users being confused and thinking they can do something that's not possible :)